backend/merge-pending.ts

```diff
@@ -0,0 +1,61 @@
+import { JSONValue } from "replicache";
+
+export async function* mergePendingChanges(
+  source: AsyncIterator<readonly [string, JSONValue], void, unknown>,
+  pending: Iterator<readonly [string, JSONValue | undefined], void>
+) {
+  for await (const [key, value] of mergePendingChangesWithDeletes(
+    source,
+    pending
+  )) {
+    if (value === undefined) {
+      continue;
+    }
+    yield [key, value] as const;
+  }
+}
+
+async function* mergePendingChangesWithDeletes(
```

Comment:
@arv we should consider adding pending to `makeScanResult` since it is something almost every impl is going to want to do. If we do it should be passed as AsyncIterator, just in case.

Reply:
Maybe we can provide a merge iterator as part of the pubic API?